gh-69929: Add more specific definition of \w (#92015)
python/cpython: Doc/library/re.rst

``\w``
   For Unicode (str) patterns:

>       Matches Unicode word characters; this includes most alphanumeric characters
>       as well as the underscore. In Unicode, alphanumeric characters are defined
>       to be the general categories L + N
>       (see `Unicode General Category Values <https://unicode.org/reports/tr44/#General_Category_Values>`_).

Reviewer: ```suggestion
      Matches Unicode word characters; this includes alphanumeric characters (as defined by :meth:`str.isalnum`)
      as well as the underscore (``_``).
```
Concrete wording suggestion
